bus/src/test/java/org.hawkular.inventory.bus/BusTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.bus;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.hawkular.bus.common.ConnectionContextFactory;
+import org.hawkular.bus.common.Endpoint;
+import org.hawkular.bus.common.MessageProcessor;
+import org.hawkular.bus.common.consumer.ConsumerConnectionContext;
+import org.hawkular.bus.common.producer.ProducerConnectionContext;
+import org.hawkular.bus.common.test.SimpleTestListener;
+import org.hawkular.bus.common.test.VMEmbeddedBrokerWrapper;
+import org.hawkular.inventory.api.Action;
+import org.hawkular.inventory.api.model.CanonicalPath;
+import org.hawkular.inventory.api.model.Environment;
+import org.hawkular.inventory.api.model.Feed;
+import org.hawkular.inventory.api.model.Metric;
+import org.hawkular.inventory.api.model.MetricDataType;
+import org.hawkular.inventory.api.model.MetricType;
+import org.hawkular.inventory.api.model.MetricUnit;
+import org.hawkular.inventory.api.model.Relationship;
+import org.hawkular.inventory.api.model.Resource;
+import org.hawkular.inventory.api.model.ResourceType;
+import org.hawkular.inventory.api.model.Tenant;
+import org.hawkular.inventory.bus.api.EnvironmentEvent;
+import org.hawkular.inventory.bus.api.FeedEvent;
+import org.hawkular.inventory.bus.api.MetricEvent;
+import org.hawkular.inventory.bus.api.MetricTypeEvent;
+import org.hawkular.inventory.bus.api.RelationshipEvent;
+import org.hawkular.inventory.bus.api.ResourceEvent;
+import org.hawkular.inventory.bus.api.ResourceTypeEvent;
+import org.hawkular.inventory.bus.api.TenantEvent;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ * @since 0.2.0
+ */
+public class BusTest {
+
+    private static final String PROP_KEY = "p1";
+    private static final String PROP_VALUE = "prop1";
+    private static final Map<String, Object> objectProperties = new HashMap<>();
+    private static final Map<String, String> messageHeaders = new HashMap<>();
+
+    @BeforeClass
+    public static void init() {
+        objectProperties.put(PROP_KEY, PROP_VALUE);
+        messageHeaders.put(PROP_KEY, PROP_VALUE);
+    }
+
+    @Test
+    public void messagesSerializationTest() {
```

Comment:
I was looking into this PR and the messages are serialized by the default object mapper. Pavol asked mazz if it'd be possible to configure own jackson object mapper for the bus-common. It should be available in the 0.3.1.Final.

Currently when the default JSON serialization is used the messages look ugly because of the path field, which uses the `toString()` representation. Also, I remember talking about it on IRC that it could worth trying to serialize the entity paths with the tenant ids, because the consumer on the bus can't figure out the tenant id.

However, this sub-task kind of complicates the original task which was only removing the gson dep., so imho we can merge and create another jira.